etl/src/energuide/embedded/basement.py

```diff
@@ -64,12 +64,17 @@ class BasementHeader(_BasementHeader):
 
     @classmethod
     def from_data(cls, header: element.Element) -> 'BasementHeader':
-        return BasementHeader(
-            nominal_insulation=insulation.Insulation(float(header.xpath('Construction/Type/@nominalInsulation')[0])),
-            effective_insulation=insulation.Insulation(float(header.xpath('Construction/Type/@rValue')[0])),
-            height=distance.Distance(float(header.xpath('Measurements/@height')[0])),
-            perimeter=distance.Distance(float(header.xpath('Measurements/@perimeter')[0])),
-        )
+        try:
+            return BasementHeader(
+                nominal_insulation=insulation.Insulation(
+                    float(header.xpath('Construction/Type/@nominalInsulation')[0])
+                ),
+                effective_insulation=insulation.Insulation(float(header.xpath('Construction/Type/@rValue')[0])),
+                height=distance.Distance(float(header.xpath('Measurements/@height')[0])),
+                perimeter=distance.Distance(float(header.xpath('Measurements/@perimeter')[0])),
+            )
+        except (ValueError, IndexError) as exc:
+            raise InvalidEmbeddedDataTypeError(BasementHeader) from exc
```

Comment:
Please add a meaningful exception message here to make it easier to Cmd-f to an answer.